transformations.html

```diff
@@ -591,6 +761,24 @@ <h2>Pattern matching</h2>
     hd::tl -> Just (hd,tl)
     []     -> Nothing
 
+
+fib n =
+    case n of
+      0 -> 1
+      1 -> 1
+      _ -> fib (n-1) + fib (n-2)
+ </code></pre>
+
+<pre><code class="gren elm">
+case maybe of
+    Just xs -> xs
+    Nothing -> []
+
+case xs of
+    []     -> Nothing
+    [a, b] -> Just { first = a, rest = [b] }
+    _ -> Array.popFirst xs
```

Comment:
This one is a little awkward, since you wouldn't do this in gren. There is no `::` and `popFirst` is already doing what the entire case expression is doing. But wrote it like this to match the syntax examples.

Reply:
@blaix I think it's fine to write it like it should be in Gren! The goal for this reference is to give an idiomatic sense too, not just be a pure conversion at the expense of sense.

And if literal conversion is fine but an alternative is preferred for some reason, I think a second example with a note is great 👍 

Leave it to you to decide.

Reply:
Changed to a more idiomatic example with a comment about lack of `::`